docker/Dockerfile

```diff
@@ -0,0 +1,291 @@
+ARG BASE_PYPI_URL
+ARG ONEAPI_INSTALL_DIR
+ARG DPNP_BUILD_DIR=$TMPDIR/dpnp
+ARG DPNP_GIT_BRANCH
+ARG DPNP_GIT_URL
+ARG DPNP_DEBUG=1
+COPY --from=dpctl_builder $TMPDIR/dpctl*.whl $TMPDIR/
+RUN pip install -U -i $INTEL_PYPI_URL --extra-index-url $BASE_PYPI_URL $TMPDIR/dpctl*.whl \
+    && mkdir -p $DPNP_BUILD_DIR \
+    && cd $DPNP_BUILD_DIR \
+    && git clone --recursive -b $DPNP_GIT_BRANCH --depth 1 $DPNP_GIT_URL . \
+    && source $ONEAPI_INSTALL_DIR/setvars.sh \
+    && export PATH="/opt/sycl/bin:$VIRTUAL_ENV/bin:$PATH" \
+    && export LD_LIBRARY_PATH="/opt/sycl/lib:$VIRTUAL_ENV/lib:$LD_LIBRARY_PATH" \
+    && export DPLROOT=$ONEAPI_ROOT/dpl/latest \
+    && python setup.py clean \
+    && export CC=dpcpp \
+    && python setup.py build_ext \
+    # TODO: manylinux wheel build arg ?
+    && python setup.py bdist_wheel \
+    && cp dist/dpctl*.whl $TMPDIR/ \
+    && rm -Rf $DPNP_BUILD_DIR
+
+FROM base_builder AS numba_dpex_builder
+SHELL ["/bin/bash", "-c"]
+ARG TMPDIR
+ARG ONEAPI_INSTALL_DIR
+ARG NUMBA_DPEX_BUILD_DIR=$TMPDIR/numba_dpex
+ARG NUMBA_DPEX_GIT_BRANCH
+ARG NUMBA_DPEX_GIT_URL
+ARG INTEL_NUMBA_VERSION
+ARG INTEL_PYPI_URL
+ARG BASE_PYPI_URL
+COPY --from=dpctl_builder $TMPDIR/dpctl*.whl $TMPDIR/
+COPY --from=dpnp_builder $TMPDIR/dpnp*.whl $TMPDIR/
+RUN pip install -U -i $INTEL_PYPI_URL --extra-index-url $BASE_PYPI_URL $TMPDIR/dpctl*.whl $TMPDIR/dpnp*.whl packaging \
+    && rm -Rf $TMPDIR/* \
+    && mkdir -p $NUMBA_DPEX_BUILD_DIR \
+    && cd $NUMBA_DPEX_BUILD_DIR \
+    && git clone --recursive -b $NUMBA_DPEX_GIT_BRANCH --depth 1 $NUMBA_DPEX_GIT_URL . \
+    && apt-get update --quiet && apt-get install -y --no-install-recommends \
+        spirv-tools spirv-headers \
+    && rm -rf /var/lib/apt/lists/* \
+    && source $ONEAPI_INSTALL_DIR/setvars.sh \
+    && export PATH=/opt/sycl/bin:$VIRTUAL_ENV/bin:$ONEAPI_INSTALL_DIR/compiler/latest/bin-llvm:$PATH \
+    && export LD_LIBRARY_PATH=/opt/sycl/lib:$VIRTUAL_ENV/lib:$LD_LIBRARY_PATH \
+    && python setup.py bdist_wheel \
+    && cp dist/numba_dpex*.whl $TMPDIR/
+
+
+# Install the 3 packages and setup the entrypoint to source the 
+# oneapi setvars.sh file on startup, and fix some other environment variables
+# accordingly.
+FROM host AS numba_dpex_dev_installer
+ARG TMPDIR
+ARG INTEL_PYPI_URL
+ARG BASE_PYPI_URL
+ARG INTEL_NUMPY_VERSION
+ARG INTEL_SCIPY_VERSION
+ARG INTEL_NUMBA_VERSION
+COPY --from=dpctl_builder $TMPDIR/dpctl*.whl $TMPDIR/
+COPY --from=dpnp_builder $TMPDIR/dpnp*.whl $TMPDIR/
+COPY --from=numba_dpex_builder $TMPDIR/numba_dpex*.whl $TMPDIR/
+RUN apt-get update --quiet && apt-get install -y --no-install-recommends spirv-tools spirv-headers \
+    && rm -Rf /var/lib/apt/lists/* \
+    && pip install -U -i $INTEL_PYPI_URL --extra-index-url $BASE_PYPI_URL "numpy==${INTEL_NUMPY_VERSION}" "numba${INTEL_NUMBA_VERSION}" "scipy${INTEL_SCIPY_VERSION}" $TMPDIR/dpctl*.whl $TMPDIR/dpnp*.whl $TMPDIR/numba_dpex*.whl cython packaging pytest pexpect \
+    && rm -Rf $TMPDIR/*
```

Comment:
I tried to build without passing any args (`docker build -t dpex .`) and got the following error:

```
WARNING: The index url "numpy====1.22.3" seems invalid, please provide a scheme.
Looking in indexes: https://pypi.anaconda.org/intel/simple, numpy====1.22.3
WARNING: Location 'numpy====1.22.3/numba/' is ignored: it is either a non-existing path or lacks a specific scheme.
Collecting numba
  Downloading https://pypi.anaconda.org/intel/simple/numba/0.54.1/numba-0.54.1-4-cp39-cp39-manylinux2014_x86_64.whl (2.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.3/2.3 MB 1.5 MB/s eta 0:00:00
WARNING: Location 'numpy====1.22.3/scipy/' is ignored: it is either a non-existing path or lacks a specific scheme.
Collecting scipy
  Downloading https://pypi.anaconda.org/intel/simple/scipy/1.7.3/scipy-1.7.3-0-cp39-cp39-manylinux_2_17_x86_64.whl (31.8 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 31.8/31.8 MB 879.5 kB/s eta 0:00:00
WARNING: Location 'numpy====1.22.3/wheel/' is ignored: it is either a non-existing path or lacks a specific scheme.
ERROR: Could not find a version that satisfies the requirement wheel (from versions: none)
ERROR: No matching distribution found for wheel
WARNING: Location 'numpy====1.22.3/pip/' is ignored: it is either a non-existing path or lacks a specific scheme.
WARNING: There was an error checking the latest version of pip.
The command '/bin/sh -c apt-get update --quiet     && apt-get install -y --no-install-recommends         build-essential         libtool         autoconf         unzip         libssl-dev         wget         ninja-build     && rm -rf /var/lib/apt/lists/*     && apt purge -y --auto-remove cmake     && mkdir -p $CMAKE_BUILD_DIR     && cd $CMAKE_BUILD_DIR     && wget https://cmake.org/files/v${CMAKE_VERSION}/cmake-${CMAKE_VERSION}.${CMAKE_VERSION_BUILD}.tar.gz     && tar -xzvf cmake-${CMAKE_VERSION}.${CMAKE_VERSION_BUILD}.tar.gz     && cd cmake-${CMAKE_VERSION}.${CMAKE_VERSION_BUILD}     && ./bootstrap     && make -j$(nproc)     && sudo make install     && rm -Rf $CMAKE_BUILD_DIR     && cd     && pip install -U -i $INTEL_PYPI_URL --extra-index-url $BASE_PYPI_URL "numpy==${INTEL_NUMPY_VERSION}" "numba${INTEL_NUMBA_VERSION}" "scipy${INTEL_SCIPY_VERSION}" wheel cython scikit-build' returned a non-zero code: 1
```